### PR TITLE
Build a rsync server docker image from infra.ci

### PR DESCRIPTION
**Pull request by olblak**
Build a rsync server docker image from infra.ci
This image will be used in the mirror helm chart, as we need a read-only rsync server to add the mirror to mirrorbits

### CODE REVIEW

**Review by garethjevans — APPROVED** [review]

File: config/default/jenkins-infra.yaml [hunk 1]
```diff
@@ -220,6 +220,23 @@ jenkins:
                     }
                   }
                 }
+            - script: >
+                multibranchPipelineJob('docker-rsyncd') {
```
Comment (timja) [inline config/default/jenkins-infra.yaml:224]: can't we just move to a github org folder? this is getting crazy updating this all the time?

**Review by markyjackson-taulia — APPROVED** [review]

File: config/default/jenkins-infra.yaml (hunk 1 above)
Comment (olblak) [reply config/default/jenkins-infra.yaml:224]: It's indeed probably a good time to switch

Comment (timja) [reply config/default/jenkins-infra.yaml:224]: here's an example of one https://github.com/hmcts/cnp-jenkins-config/blob/master/jobdsl/organisations-beta.groovy

**Comment by olblak** [comment]
```
grafana, grafana, StatefulSet (apps) has changed:
  # Source: grafana/templates/statefulset.yaml
  apiVersion: apps/v1
  kind: StatefulSet
  metadata:
    name: grafana
    namespace: grafana
    labels:
      helm.sh/chart: grafana-6.1.16
      app.kubernetes.io/name: grafana
      app.kubernetes.io/instance: grafana
      app.kubernetes.io/version: "7.3.5"
      app.kubernetes.io/managed-by: Helm
  spec:
    replicas: 1
    selector:
      matchLabels:
        app.kubernetes.io/name: grafana
        app.kubernetes.io/instance: grafana
    serviceName: grafana-headless
    template:
      metadata:
        labels:
          app.kubernetes.io/name: grafana
          app.kubernetes.io/instance: grafana
        annotations:
          checksum/config: 39cee3da168106b8b5a3c50f1952687357f8d34661dffa4cd94cc76f52491ca9
          checksum/dashboards-json-config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b
          checksum/sc-dashboard-provider-config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b
-         checksum/secret: c50dd40b0f22f18007aad582727811994508a64a7028c4f94b95cee00e6ebf67
+         checksum/secret: a7bf34d22aba1c7df816081ff7b164743ae01f1d8afd67196a1d9cfbdae3e8db
      spec:
        
        serviceAccountName: grafana
        securityContext:
          fsGroup: 472
          runAsGroup: 472
          runAsUser: 472
        initContainers:
          - name: init-chown-data
            image: "busybox:1.31.1"
            imagePullPolicy: IfNotPresent
            securityContext:
              runAsNonRoot: false
              runAsUser: 0
            command: ["chown", "-R", "472:472", "/var/lib/grafana"]
            resources:
              {}
            volumeMounts:
              - name: storage
                mountPath: "/var/lib/grafana"
        containers:
          - name: grafana
            image: "grafana/grafana:7.3.5"
            imagePullPolicy: IfNotPresent
            volumeMounts:
              - name: config
                mountPath: "/etc/grafana/grafana.ini"
                subPath: grafana.ini
              - name: ldap
                mountPath: "/etc/grafana/ldap.toml"
                subPath: ldap.toml
              - name: storage
                mountPath: "/var/lib/grafana"
              - name: config
                mountPath: "/etc/grafana/provisioning/datasources/datasources.yaml"
                subPath: datasources.yaml
            ports:
              - name: service
                containerPort: 80
                protocol: TCP
              - name: grafana
                containerPort: 3000
                protocol: TCP
            env:
              - name: GF_SECURITY_ADMIN_USER
                valueFrom:
                  secretKeyRef:
                    name: grafana
                    key: admin-user
              - name: GF_SECURITY_ADMIN_PASSWORD
                valueFrom:
                  secretKeyRef:
                    name: grafana
                    key: admin-password
              
            livenessProbe:
              failureThreshold: 10
              httpGet:
                path: /api/health
                port: 3000
              initialDelaySeconds: 60
              timeoutSeconds: 30
            readinessProbe:
              httpGet:
                path: /api/health
                port: 3000
            resources:
              limits:
                cpu: 200m
                memory: 256Mi
              requests:
                cpu: 100m
                memory: 128Mi
        volumes:
          - name: config
            configMap:
              name: grafana
          - name: ldap
            secret:
              secretName: grafana
              items:
                - key: ldap-toml
                  path: ldap.toml
        # nothing
    volumeClaimTemplates:
    - metadata:
        name: storage
      spec:
        accessModes: [ReadWriteOnce]
        storageClassName: 
        resources:
          requests:
            storage: 50
grafana, grafana, Secret (v1) has changed:
+ Changes suppressed on sensitive content of type Secret

jenkins-infra, jenkins-infra-jenkins-config-jobs-settings, ConfigMap (v1) has changed:
  # Source: jenkins/charts/jenkins/templates/jcasc-config.yaml
  apiVersion: v1
  kind: ConfigMap
  metadata:
    name: jenkins-infra-jenkins-config-jobs-settings
    namespace: jenkins-infra
    labels:
      "app.kubernetes.io/name": jenkins
      "helm.sh/chart": "jenkins-2.19.0"
      "app.kubernetes.io/managed-by": "Helm"
      "app.kubernetes.io/instance": "jenkins-infra"
      "app.kubernetes.io/component": "jenkins-master"
      jenkins-infra-jenkins-config: "true"
  data:
    jobs-settings.yaml: |-
      jobs:
        - script: >
            multibranchPipelineJob('k8smgmt') {
              displayName "K8s Cluster Management"
              description "Current K8s Cluster Management"
              branchSources {
                github {
                  id('2019081602')
                  scanCredentialsId('github-access-token')
                  repoOwner('jenkins-infra')
                  repository('charts')
                }
              }
              factory {
                workflowBranchProjectFactory {
                  scriptPath('Jenkinsfile_k8s')
                }
              }
              configure { node ->
                def traits = node / 'sources' / 'data' / 'jenkins.branch.BranchSource' / 'source' / 'traits'
                traits << 'org.jenkinsci.plugins.github__branch__source.BranchDiscoveryTrait' {
                  strategyId(1)
                }
                traits << 'org.jenkinsci.plugins.github__branch__source.OriginPullRequestDiscoveryTrait' {
                  strategyId(1)
                }
                traits << 'org.jenkinsci.plugins.github__branch__source.ForkPullRequestDiscoveryTrait' {
                  strategyId(1)
                  trust(class: 'org.jenkinsci.plugins.github_branch_source.ForkPullRequestDiscoveryTrait$TrustPermission')
                }
                traits << 'jenkins.scm.impl.trait.WildcardSCMHeadFilterTrait' {
                  includes('main master PR-*')
                  excludes()
                }
                traits << 'org.jenkinsci.plugins.github.label.filter.PullRequestLabelsBlackListFilterTrait' {
                  labels('on-hold ci-skip')
                }
              }
            }
        - script: >
            multibranchPipelineJob('plugin-site') {
              displayName "Plugin Site"
              branchSources {
                github {
                  id('2019081602')
                  scanCredentialsId('github-access-token')
                  repoOwner('jenkins-infra')
                  repository('plugin-site')
                  includes('master')
                }
              }
              factory {
                workflowBranchProjectFactory {
                  scriptPath('Jenkinsfile_k8s')
                }
              }
            }
        - script: >
            multibranchPipelineJob('docker-packer') {
              displayName "Packer Docker Build"
              branchSources {
                github {
                  id('2020101901')
                  scanCredentialsId('github-access-token')
                  repoOwner('jenkins-infra')
                  repository('docker-packer')
                }
              }
              factory {
                workflowBranchProjectFactory {
                  scriptPath('Jenkinsfile_k8s')
                }
              }
            }
        - script: >
            multibranchPipelineJob('docker-mirrorbits') {
              displayName "Mirrorbits Docker Build"
              branchSources {
                github {
                  id('2020102301')
                  scanCredentialsId('github-access-token')
                  repoOwner('jenkins-infra')
                  repository('docker-mirrorbits')
                }
              }
              factory {
                workflowBranchProjectFactory {
                  scriptPath('Jenkinsfile_k8s')
                }
              }
            }
        - script: >
            multibranchPipelineJob('docker-datadog') {
              displayName "DataDog Docker Build"
              branchSources {
                github {
                  id('2020111801')
                  scanCredentialsId('github-access-token')
                  repoOwner('jenkins-infra')
                  repository('docker-datadog')
                }
              }
              factory {
                workflowBranchProjectFactory {
                  scriptPath('Jenkinsfile_k8s')
                }
              }
            }
        - script: >
+           multibranchPipelineJob('docker-rsyncd') {
+             displayName "Rsyncd Docker Build"
+             branchSources {
+               github {
+                 id('2021010601')
+                 scanCredentialsId('github-access-token')
+                 repoOwner('jenkins-infra')
+                 repository('docker-rsyncd')
+               }
+             }
+             factory {
+               workflowBranchProjectFactory {
+                 scriptPath('Jenkinsfile_k8s')
+               }
+             }
+           }
+       - script: >
            multibranchPipelineJob('docker-terraform') {
              displayName "Terraform Docker Build"
              branchSources {
                github {
                  id('2020121001')
                  scanCredentialsId('github-access-token')
                  repoOwner('jenkins-infra')
                  repository('docker-terraform')
                }
              }
              factory {
                workflowBranchProjectFactory {
                  scriptPath('Jenkinsfile_k8s')
                }
              }
            }
        - script: >
            multibranchPipelineJob('docker-builder') {
              displayName "Docker Builder's Docker Build"
              branchSources {
                github {
                  id('2020122201')
                  scanCredentialsId('github-access-token')
                  repoOwner('jenkins-infra')
                  repository('docker-builder')
                }
              }
              factory {
                workflowBranchProjectFactory {
                  scriptPath('Jenkinsfile_k8s')
                }
              }
            }
        - script: >
            multibranchPipelineJob('aws') {
              displayName "AWS Infra"
              branchSources {
                github {
                  id('2020120401')
                  scanCredentialsId('github-access-token')
                  repoOwner('jenkins-infra')
                  repository('aws')
                }
              }
              factory {
                workflowBranchProjectFactory {
                  scriptPath('Jenkinsfile_k8s')
                }
              }
            }
        - script: >
            [
              ['jenkins-infra', 'jenkins-wiki-exporter', 'Wiki Exporter'],
              ['jenkinsci', 'custom-distribution-service', 'Custom Distribution Service'],
              ['jenkins-infra', 'incrementals-publisher', 'Incrementals Publisher'],
            ].each { config ->
              multibranchPipelineJob(config[1]) {
                displayName config[2]
                triggers {
                  pollSCM {
                    // poll every two hours until hooks are setup
                    scmpoll_spec('H/2 * * * *')
                  }
                }
                branchSources {
                  branchSource {
                    source {
                      github {
                        credentialsId("github-access-token")
                        configuredByUrl(true)
                        repositoryUrl('https://github.com/' + config[0] + '/' + config[1])
                        repoOwner(config[0])
                        repository(config[1])
                        traits {
                          gitHubBranchDiscovery {
                            strategyId(1) // 1-only branches that are not pull requests
                          }
                          pruneStaleBranchTrait()
                          gitHubTagDiscovery()
                        }
                      }
                      buildStrategies {
                        buildAnyBranches {
                          strategies {
                            buildAllBranches {
                              strategies {
                                buildNamedBranches {
                                  filters {
                                    exact {
                                      name("master")
                                      caseSensitive(true)
                                    }
                                  }
                                }
                              }
                            }
                          }
                        }
                        buildAnyBranches {
                          strategies {
                            buildAllBranches {
                              strategies {
                                buildTags {
                                  atLeastDays '-1'
                                  atMostDays '3'
                                }
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
                factory {
                  workflowBranchProjectFactory {
                    scriptPath('Jenkinsfile')
                  }
                }
              }
            }
jenkins-infra, jenkins-infra, Secret (v1) has changed:
+ Changes suppressed on sensitive content of type Secret

release, default-release-jenkins, Secret (v1) has changed:
+ Changes suppressed on sensitive content of type Secret
release, default-release-jenkins-jenkins-config-jobs-settings, ConfigMap (v1) has changed:
  # Source: jenkins/charts/jenkins/templates/jcasc-config.yaml
  apiVersion: v1
  kind: ConfigMap
  metadata:
    name: default-release-jenkins-jenkins-config-jobs-settings
    namespace: release
    labels:
      "app.kubernetes.io/name": jenkins
      "helm.sh/chart": "jenkins-2.19.0"
      "app.kubernetes.io/managed-by": "Helm"
      "app.kubernetes.io/instance": "default-release-jenkins"
      "app.kubernetes.io/component": "jenkins-master"
      default-release-jenkins-jenkins-config: "true"
  data:
    jobs-settings.yaml: |-
      jobs:
        - script: >
            folder('components') {
              displayName('Components')
              description('Folder containing Jenkins components release job')
            }
      
        - script: >
            multibranchPipelineJob('components/remoting') {
              displayName "Remoting"
              description "Jenkins Remoting"
              branchSources {
                github {
                  id('2020052601')
                  scanCredentialsId('github-access-token')
                  repoOwner('jenkins-infra')
                  repository('release')
                  includes('master')
                }
              }
              factory {
                workflowBranchProjectFactory {
                  scriptPath('Jenkinsfile.d/components/remoting')
                }
              }
              configure {
                it / sources / 'data' / 'jenkins.branch.BranchSource' << {
                    strategy(class: 'jenkins.branch.DefaultBranchPropertyStrategy') {
                        properties(class: 'java.util.Arrays$ArrayList') {
                            a(class: 'jenkins.branch.BranchProperty-array') {
                                'jenkins.branch.NoTriggerBranchProperty'()
                            }
                        }
                    }
                }
              }
            }
      
        - script: >
            folder('core') {
              displayName('Core')
              description('Folder containing Jenkins core release job')
            }
      
        - script: >
            folder('core/weekly') {
              displayName('Weekly')
              description('Folder for weekly releases')
            }
      
        - script: >
            folder('core/stable') {
              displayName('Stable')
              description('Folder for stable releases')
            }
        - script: >
            folder('core/stable-rc') {
              displayName('Stable Release Candidates')
              description('Folder for stable release candidates')
            }
      
        - script: >
            multibranchPipelineJob('core/package') {
              displayName "Core Package"
              description "Jenkins Core Packaging"
              branchSources {
                github {
                  id('2019092401')
                  scanCredentialsId('github-access-token')
                  repoOwner('jenkins-infra')
                  repository('release')
                }
              }
              factory {
                workflowBranchProjectFactory {
                  scriptPath('Jenkinsfile.d/core/package')
                }
              }
              configure {
                it / sources / 'data' / 'jenkins.branch.BranchSource' << {
                    strategy(class: 'jenkins.branch.DefaultBranchPropertyStrategy') {
                        properties(class: 'java.util.Arrays$ArrayList') {
                            a(class: 'jenkins.branch.BranchProperty-array') {
                                'jenkins.branch.NoTriggerBranchProperty'()
                            }
                        }
                    }
                }
              }
            }
        - script: >
            multibranchPipelineJob('core/release') {
              displayName "Core Release"
              description "Jenkins Core Release"
              branchSources {
                github {
                  id('2019092402')
                  scanCredentialsId('github-access-token')
                  repoOwner('jenkins-infra')
                  repository('release')
                }
              }
              factory {
                workflowBranchProjectFactory {
                  scriptPath('Jenkinsfile.d/core/release')
                }
              }
              configure {
                it / sources / 'data' / 'jenkins.branch.BranchSource' << {
                    strategy(class: 'jenkins.branch.DefaultBranchPropertyStrategy') {
                        properties(class: 'java.util.Arrays$ArrayList') {
                            a(class: 'jenkins.branch.BranchProperty-array') {
                                'jenkins.branch.NoTriggerBranchProperty'()
                            }
                        }
                    }
                }
              }
            }
      
        - script: >
            multibranchPipelineJob('core/weekly/release') {
              displayName "Release"
              description "Jenkins Core Release"
              branchSources {
                github {
                  id('2019092402')
                  scanCredentialsId('github-access-token')
                  repoOwner('jenkins-infra')
                  repository('release')
                  includes('master')
                }
              }
              factory {
                workflowBranchProjectFactory {
                  scriptPath('Jenkinsfile.d/core/weekly')
                }
              }
              configure {
                it / sources / 'data' / 'jenkins.branch.BranchSource' << {
                    strategy(class: 'jenkins.branch.DefaultBranchPropertyStrategy') {
                        properties(class: 'java.util.Arrays$ArrayList') {
                            a(class: 'jenkins.branch.BranchProperty-array') {
                                'jenkins.branch.NoTriggerBranchProperty'()
                            }
                        }
                    }
                }
              }
            }
        - script: >
            multibranchPipelineJob('core/stable/release') {
              displayName "Release"
              description "Jenkins Core Release"
              branchSources {
                github {
                  id('2019092402')
                  scanCredentialsId('github-access-token')
                  repoOwner('jenkins-infra')
                  repository('release')
                }
              }
              factory {
                workflowBranchProjectFactory {
                  scriptPath('Jenkinsfile.d/core/stable')
                }
              }
              configure {
                it / sources / 'data' / 'jenkins.branch.BranchSource' << {
                    strategy(class: 'jenkins.branch.DefaultBranchPropertyStrategy') {
                        properties(class: 'java.util.Arrays$ArrayList') {
                            a(class: 'jenkins.branch.BranchProperty-array') {
                                'jenkins.branch.NoTriggerBranchProperty'()
                            }
                        }
                    }
                }
              }
            }
        - script: >
            multibranchPipelineJob('core/stable-rc/release') {
              displayName "Release"
              description "Jenkins Core Release"
              branchSources {
                github {
-                 id('2020121601')
+                 id('2020121702')
                  scanCredentialsId('github-access-token')
                  repoOwner('jenkins-infra')
                  repository('release')
                }
              }
              factory {
                workflowBranchProjectFactory {
                  scriptPath('Jenkinsfile.d/core/stable-rc')
                }
              }
              configure {
                it / sources / 'data' / 'jenkins.branch.BranchSource' << {
                    strategy(class: 'jenkins.branch.DefaultBranchPropertyStrategy') {
                        properties(class: 'java.util.Arrays$ArrayList') {
                            a(class: 'jenkins.branch.BranchProperty-array') {
                                'jenkins.branch.NoTriggerBranchProperty'()
                            }
                        }
                    }
                }
              }
            }

File: config/default/jenkins-infra.yaml (hunk 1 above)
Comment (garethjevans) [reply config/default/jenkins-infra.yaml:224]: I've pushed https://github.com/jenkins-infra/charts/pull/733, as an attempt to configure the `jenkins-infra/docker-.*` repos in this way

**Comment by garethjevans** [comment]
@olblak I think this can be closed in favour of https://github.com/jenkins-infra/charts/pull/733

**Pull request closed without merging**